### PR TITLE
Document steps to deploy P256 verification contract at predetermined address

### PR DESCRIPTION
**Pull request by gabspeck**
This PR was originally opened to change the P256 verifier address to the official precompile specified by RIP-7212, however we hit two issues that paused that:

* Hardhat claims to support it via a config flag, but it seems not to be working
* Not all EVM implementations support it yet. Case in point: Besu.

So for now we are sticking to the currently used address and documenting how to deploy it on new chains. For the future, this deployment should be automated using the Hardhat toolkit. This documentation is more of a reminder on how to do this in the future if needed.

As a side effect of the investigation, I've also added an import of the `hardhat-tracer` plugin that helps to debug mysterious Hardhat reverts with no reason strings.

### CODE REVIEW

File: DEPLOY_VERIFIER.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        // The salt used (can be any bytes32 value)
+        bytes32 salt = bytes32(0); // Using zero salt as per your requirements
```
Comment (kozaqui) [inline DEPLOY_VERIFIER.md:56]: probs this comment can be changed
`Using zero salt as per your requirements`

File: DEPLOY_VERIFIER.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+These steps are only necessary if your chain does NOT have the factory deployed at `0x4e59b44847b379578588920ca78fbf26c0b4956c`
+1. Download and install Foundry
+2. Fund the account that signed the factory deployment transaction:
+
+```sh
+cast send --value 1000000000gwei --private-key <key with ETH> 0x3fab184622dc19b6109349b94811493bf2a45362 --rpc-url <RPC endpoint URL>`
+```
+3. Publish the pre-signed deployment transaction:
+```sh
```
Comment (kozaqui) [inline DEPLOY_VERIFIER.md:26]: Could add a newline before the codeblock

File: DEPLOY_VERIFIER.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        // Extract the deployed contract address from returnData
+        address deployedAddress;
+        assembly {
+            // Skip the length prefix (first 32 bytes) and the first 12 bytes (offset) to get the address
+            deployedAddress := mload(add(returnData, 32))
+        }
+
+        vm.stopBroadcast();
+
+        console.log("Deployed contract at:", deployedAddress);    }
```
Comment (kozaqui) [inline DEPLOY_VERIFIER.md:74]: hah, that's a nice thing to have in Solidity, good to know that ;)

**Review by kozaqui — APPROVED** [review]
A very solid investigation + documentation 🫡

File: DEPLOY_VERIFIER.md (hunk 1 above)
Comment (gabspeck) [reply DEPLOY_VERIFIER.md:56]: Hah, nice catch, ChatGPT leftover

**Pull request merged**